app/jobs/inat_import_job.rb

```diff
@@ -0,0 +1,408 @@
+    # To get one page, use iNats `per_page` & `id_above` params.
+    # https://api.inaturalist.org/v1/docs/#!/Observations/get_observations
+    last_import_id = 0
+    loop do
+      page_of_observations =
+        # get a page of observations with id > id of last imported obs
+        next_page(id: inat_ids, id_above: last_import_id,
+                  user_login: @inat_import.inat_username)
+      return if JSON.parse(page_of_observations.body)["error"].present?
+
+      parsed_page = JSON.parse(page_of_observations)
+      @inat_import.update(importables: parsed_page["total_results"])
+      break if page_empty?(parsed_page)
+
+      import_page(page_of_observations)
+
+      last_import_id = parsed_page["results"].last["id"]
+      next unless last_page?(parsed_page)
+
+      break
+    end
+  end
+
+  def page_empty?(page)
+    page["total_results"].zero?
+  end
+
+  def last_page?(parsed_page)
+    parsed_page["total_results"] <=
+      parsed_page["page"] * parsed_page["per_page"]
+  end
+
+  def inat_id_list
+    @inat_import.inat_ids.delete(" ")
+  end
+
+  # This is where we actually hit the iNat API
+  # https://api.inaturalist.org/v1/docs/#!/Observations/get_observations
+  # https://stackoverflow.com/a/11251654/3357635
+  # Note that the `ids` parameter may be a comma-separated list of iNat obs
+  # ids - that needs to be URL encoded to a string when passed as an arg here
+  # because URI.encode_www_form deals with arrays by passing the same key
+  # multiple times.
+  def next_page(**args)
+    query_args = {
+      id: nil, id_above: nil, only_id: false, per_page: 200,
+      order: "asc", order_by: "id",
+      # prevents user from importing others' obss
+      user_login: nil, iconic_taxa: ICONIC_TAXA
+    }.merge(args)
+
+    query = URI.encode_www_form(query_args)
+
+    # ::Inat.new(operation: query, token: @inat_import.token).body
+    # Nimmo 2024-06-19 jdc. Moving the request from the inat class to here.
+    # RestClient::Request.execute wasn't available in the class
+    headers = { authorization: "Bearer #{@inat_import.token}", accept: :json }
+    @inat = RestClient::Request.execute(
+      method: :get, url: "#{API_BASE}/observations?#{query}", headers: headers
+    )
+  rescue RestClient::ExceptionWithResponse => e
+    e.response
+  end
+
+  def import_page(page)
+    JSON.parse(page)["results"].each do |result|
+      import_one_result(JSON.generate(result))
+    end
+  end
+
+  def import_one_result(result)
+    @inat_obs = Inat::Obs.new(result)
+    return unless @inat_obs.importable?
+
+    create_observation
+    add_inat_images(@inat_obs.inat_obs_photos)
+    update_names_and_proposals
+    add_inat_sequences
+    add_snapshot_of_import_comment
+    # NOTE: update field slip 2024-09-09 jdc
+    # https://github.com/MushroomObserver/mushroom-observer/issues/2380
+    update_inat_observation
+    increment_imported_count
+  end
+
+  def create_observation
```

Comment:
For this ABC metric violation, I consider implementing something like `create_hash` on the the `Inat::Obs` that returns a hash which you could then add in the other stuff with a merge and pass the resulting to `create` rather than the big long list of key value pairs.